src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMSource.java

```
@@ -0,0 +1,445 @@
+    }
+
+    @Override
+    public SCM build(SCMHead head, @CheckForNull SCMRevision revision) {
+        if (LOGGER.isLoggable(Level.FINE)) {
+            LOGGER.fine("Building SCM for " + head.getName() + " at revision " + revision);
+        }
+        return gitSCMSource.build(head, revision);
+    }
+
+    public BitbucketSCMRepository getBitbucketSCMRepository() {
+        return repository;
+    }
+
+    @CheckForNull
+    public String getCredentialsId() {
+        return gitSCMSource.getCredentialsId();
+    }
+
+    public String getMirrorName() {
+        return getBitbucketSCMRepository().getMirrorName();
+    }
+
+    public String getProjectKey() {
+        return getBitbucketSCMRepository().getProjectKey();
+    }
+
+    public String getProjectName() {
+        return getBitbucketSCMRepository().getProjectName();
+    }
+
+    public String getRemote() {
+        return gitSCMSource.getRemote();
+    }
+
+    public String getRepositoryName() {
+        return getBitbucketSCMRepository().getRepositoryName();
+    }
+
+    public String getRepositorySlug() {
+        return getBitbucketSCMRepository().getRepositorySlug();
+    }
+
+    @CheckForNull
+    public String getServerId() {
+        return getBitbucketSCMRepository().getServerId();
+    }
+
+    @Override
+    public List<SCMSourceTrait> getTraits() {
+        return traits;
+    }
+
+    @Override
+    protected void retrieve(@CheckForNull SCMSourceCriteria criteria, SCMHeadObserver observer, @CheckForNull SCMHeadEvent<?> event, TaskListener listener) throws IOException, InterruptedException {
+        gitSCMSource.accessibleRetrieve(criteria, observer, event, listener);
+    }
+
+    private String getCloneUrl(List<BitbucketNamedLink> cloneUrls) {
+        return cloneUrls.stream()
+                .filter(link -> "http".equals(link.getName()))
+                .findFirst()
+                .map(BitbucketNamedLink::getHref)
+                .orElse("");
```

without knowing all the context: is retuning an empty string the best we can do here? What are the consequences of an empty string being passed into `new UserRemoteConfig(...)`?
I don't know the usual Jenkins way/conventions for this sort of case; will this cause something to explode down the track, and if yes, is it worth failing early here with proper logging and error-handling?

---

We can log, but if we throw we will prevent the saving of the multibranch config, which is probably not really what we want to do.